recipe_scrapers/kptncook.py

```diff
@@ -0,0 +1,121 @@
+import json
+from urllib.parse import parse_qs, urlparse
+
+import requests
+
+from recipe_scrapers.settings import settings
+
+from ._abstract import AbstractScraper
+
+HEADERS = {
+    "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0"
+}
+# These languages use measurements like cups and spoons rather than grams and liters
+IMPERIAL_LANGUAGES = ["en"]
+KPTN_DEFAULT_LANGUAGE = "en"
+
+
+class KptnCook(AbstractScraper):
+    def __init__(self, url, *args, **kwargs):
+        if settings.TEST_MODE:
+            self.recipe_json = json.loads(url.read())[0]
+            self.lang = KPTN_DEFAULT_LANGUAGE
+            self.final_url = "https://mobile.kptncook.com/recipe/pinterest/Low-Carb-Tarte-Flamb%C3%A9e-with-Serrano-Ham-%26-Cream-Cheese/315c3c32?lang=en"
+        else:
+            if urlparse(url).hostname == "mobile.kptncook.com":
+                parsed_url = urlparse(url)
+            else:
+                # If it's a sharing.kptncook.com/* link we first need to follow the http forwards to get the recipe ID
+                recipe_request = requests.get(url, headers=HEADERS)
```

Comment:
A potential problem here: I think this means that we will follow the redirect twice for `sharing.` recipes.

The first redirect will occur during this GET request: https://github.com/hhursev/recipe-scrapers/blob/9f7288f3f459cc5a87aff5a50600b90eaa0c32a5/recipe_scrapers/_abstract.py#L33-L35

(the `requests` library currently [follows redirects by default](https://github.com/psf/requests/blob/da9996fe4dc63356e9467d0a5e10df3d89a8528e/requests/api.py#L36))

...and then this subsequent call will also follow the redirect, causing a bit of unnecessary server-side traffic.

I think we should try to figure out how to capture the relevant destination URL in the base class.

Reply:
I tried to implement this but in my testing the url doesn't show the final redirected url. Did I get something wrong?
```python
>>> import requests
>>> resp = requests.get("https://sharing.kptncook.com/VQ83EqYvomb", allow_redirects=True)
>>> resp.url
'https://sharing.kptncook.com/VQ83EqYvomb'
```

Reply:
Ok I did some further testing, after setting the HEADERS it works.

```python
>>> HEADERS = {
...     "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:86.0) Gecko/20100101 Firefox/86.0"
... }
>>> resp  = requests.get("https://sharing.kptncook.com/VQ83EqYvomb", headers=HEADERS, allow_redirects=True)
>>> resp.url
'http://mobile.kptncook.com/recipe/pinterest/Low-Carb-Tarte-Flamb%C3%A9e-with-Serrano-Ham-%26-Cream-Cheese/315c3c32?_branch_match_id=1070992230709572185&utm_source=Other&utm_medium=sharing&_branch_referrer=H4sIAAAAAAAAA8soKSkottLXL85ILMrMS9fLLijJS87Pz9ZLzs%2FVDwu0MHYtjCzLz00CANP8RXcoAAAA'
```